### PR TITLE
release: v1.34.0

### PR DESCRIPTION
**Pull request by goto-bus-stop**
> **Note**
> **This particular PR should be true-merged to `main`.**

This PR represents the merge to `main` of the v1.34.0 release.

This PR is **primarily a merge commit**, so reviewing every individual commit shown below is **not necessary** since those have been reviewed in their own PR.

**However!** Some things to review on this PR:

- Does this PR target the right branch? (usually, `main`)
- Are the appropriate **version bumps** and **release note edits** in the end of the commit list (or within the last few commits).  In other words, "Did the 'release prep' PR actually land on this branch?"

If those things look good, this PR is good to merge.

### CODE REVIEW

**Comment by goto-bus-stop** [comment]
I think the merge conflicts are just because the 1.33.2 release was squash-merged into dev here: https://github.com/apollographql/router/pull/4112

I can resolve them here and then it should be fine in the future after the next merge of main into dev

**Review by garypen — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,206 @@ All notable changes to Router will be documented in this file.
...
+   - altering the priority of metrics submission vs metrics gathering
+ - switching to use a `tokio` bounded channel which enforces the bound as we originally expected
+
+With these changes in place we have observed that the router behaves very well with respect to memory consumption under high load.
+
+By [@garypen](https://github.com/garypen) in https://github.com/apollographql/router/pull/4107
+
+### Support authorization directive renaming ([PR #3949](https://github.com/apollographql/router/pull/3949))
+
+When importing directives through the `@link` directive, they can be renamed. This makes sure that the authorization plugin can still recognize its directives when they have been renamed.
```
Comment (abernix) [inline CHANGELOG.md:131]:
```suggestion
When importing directives into subgraph schemas using the `@link` directive it is possible to rename the directive [using the `as:` parameter](https://specs.apollo.dev/link/v1.0/#@link.as). It is now possible to do this for authorization directives and they will be recognized appropriately even if they have been renamed.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,206 @@ All notable changes to Router will be documented in this file.
...
+
+By [@Geal](https://github.com/Geal) in https://github.com/apollographql/router/pull/3949
+
+## 📃 Configuration
+
+### Bring telemetry tracing config and metrics config into alignment ([Issue #4043](https://github.com/apollographql/router/issues/4043))
+
+Configuration between tracing and metrics was inconsistent and did not align with otel spec terminology. The following changes have been made to router.yaml configuration:
```
Comment (abernix) [inline CHANGELOG.md:139]:
```suggestion
Configuration between tracing and metrics was inconsistent and did not align with the terminology defined in the OpenTelemetry specification. To correct this, the following changes have been made to the router's YAML configuration:
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,206 @@ All notable changes to Router will be documented in this file.
...
++      aws_xray: true
+```
+
+The Router will upgrade any existing configuration on startup. However, you should update your configuration to use the new format as soon as possible. 
+
+By [@BrynCooke](https://github.com/BrynCooke) in https://github.com/apollographql/router/pull/4044, https://github.com/apollographql/router/pull/4050 and https://github.com/apollographql/router/pull/4051
+
+## 🛠 Maintenance
+
+### Router should respond with subscription-protocol header for callback ([Issue #3929](https://github.com/apollographql/router/issues/3929))
```
Comment (abernix) [inline CHANGELOG.md:182]:
```suggestion
### Router should respond with `subscription-protocol` header for callback ([Issue #3929](https://github.com/apollographql/router/issues/3929))
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,206 @@ All notable changes to Router will be documented in this file.
...
+
+Callback protocol documentation specifies that router responds with `subscription-protocol: callback/1.0` header to the initialization (check) message. Currently router does not set this header on the response.
```
Comment (abernix) [inline CHANGELOG.md:184]:
```suggestion
The router will now include a `subscription-protocol: callback/1.0` header on the response to a initialization (check) message, per the [callback protocol documentation](https://www.apollographql.com/docs/router/executing-operations/subscription-callback-protocol).
```

@bnjjj Note that the [docs](https://www.apollographql.com/docs/router/executing-operations/subscription-callback-protocol) now differ from [the `dev-docs`](https://github.com/apollographql/router/blob/dev/dev-docs/callback_protocol.md)!   Can you fix that?

**Review by abernix — COMMENTED** [review]
I've left a few comments on the changelog, but I would like @shorgi to take an overall look and also for @chandrikas to note/review the details of the Enterprise features.

File: CHANGELOG.md (hunk 4 above)
Comment (abernix) [reply CHANGELOG.md]: Covered by https://github.com/apollographql/router/issues/3937 already.  Thanks!

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,206 @@ All notable changes to Router will be documented in this file.
...
+
+Handling of Otel `service.name` has been brought into line with the [Otel spec](https://opentelemetry.io/docs/concepts/sdk-configuration/general-sdk-configuration/#otel_service_name) across traces and metrics.
+
+Service name discovery is handled in the following order:
+1. `OTEL_SERVICE_NAME` env
+2. `OTEL_RESOURCE_ATTRIBUTES` env
+3. `router.yaml` `service_name`
+4. `router.yaml` `resources` (attributes)
+
+If none of the above are found then the service name will be set to `unknown_service:apollo_router` or `unknown_service` if the executable name cannot be determined.
```
[the remark this answers is not in the document]
Comment (abernix) [reply CHANGELOG.md:110]: Ok, I guess that's probably `router` for us then?:

```suggestion
If none of the above are found then the service name will be set to `unknown_service:router` or `unknown_service` if the executable name cannot be determined.
```

File: docs/source/configuration/metrics.mdx [hunk 6]
```diff
@@ -4,18 +4,48 @@ title: Collecting metrics in the Apollo Router
...
+      service_name: "router"
+```
+
+Service name discovery is handled in the following order:
+1. `OTEL_SERVICE_NAME` env
+2. `OTEL_RESOURCE_ATTRIBUTES` env
+3. `router.yaml` `service_name`
+4. `router.yaml` `resource`
+
+If none of the above are found then the service name will be set to `unknown_service:apollo_router` or `unknown_service` if the executable name cannot be determined.
```
Comment (BrynCooke) [inline docs/source/configuration/metrics.mdx:25]:
```suggestion
If none of the above are found then the service name will be set to `unknown_service:router` or `unknown_service` if the executable name cannot be determined.
```

File: docs/source/configuration/tracing.mdx [hunk 7]
```diff
@@ -24,39 +24,54 @@ Span data is sent to a collector such as [Jaeger](https://www.jaegertracing.io/)
...
+3. `router.yaml` `service_name`
+4. `router.yaml` `resource`
 
-      # Optional limits
-      max_attributes_per_event: 10
-      max_attributes_per_link: 10
-      max_attributes_per_span: 10
-      max_events_per_span: 10
-      max_links_per_span: 10
+If none of the above are found then the service name will be set to `unknown_service:apollo_router` or `unknown_service` if the executable name cannot be determined.
```
Comment (BrynCooke) [inline docs/source/configuration/tracing.mdx:43]:
```suggestion
If none of the above are found then the service name will be set to `unknown_service:router` or `unknown_service` if the executable name cannot be determined.
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -4,6 +4,219 @@ All notable changes to Router will be documented in this file.
...
+
+Handling of OpenTelemetry (OTel) `service.name` has been brought into line with the [OTel specification](https://opentelemetry.io/docs/concepts/sdk-configuration/general-sdk-configuration/#otel_service_name) across traces and metrics.
+
+Service name discovery is handled in the following order:
+1. `OTEL_SERVICE_NAME` env
+2. `OTEL_RESOURCE_ATTRIBUTES` env
+3. `router.yaml` `service_name`
+4. `router.yaml` `resources` (attributes)
+
+If none of the above are found then the service name will be set to `unknown_service:apollo_router` or `unknown_service` if the executable name cannot be determined.
```
Comment (BrynCooke) [inline CHANGELOG.md]:
```suggestion
If none of the above are found then the service name will be set to `unknown_service:router` or `unknown_service` if the executable name cannot be determined.
```

**Review by BrynCooke — CHANGES_REQUESTED** [review]
Default OpenTelemetry service name fixes

**Review by BrynCooke — APPROVED** [review]

**Pull request merged**